src/main/scala/com/github/krasserm/ases/EventSourcing.scala

```diff
@@ -227,9 +228,9 @@ private class EventSourcing[S, E, REQ, RES](
         tryPullEi()
 
       private def tryPullEi(): Unit =
-        if (!requestUpstreamFinished) pull(ei)
+        if (!requestUpstreamFinished && !hasBeenPulled(ei) && !isClosed(ei)) pull(ei)
```

Comment:
I think this change is not needed. `tryPullEi` is called `onPush` within `ei` handler, hence `!isClosed` and `!hasBeenPulled` always evaluate to `true` and can therefore be removed.